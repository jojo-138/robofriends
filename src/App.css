.App {
  text-align: center;
}

.title {
  font-family: 'Alumni Sans Inline One', cursive;
  margin-top: 3vh;
  margin-bottom: 3vh;
}