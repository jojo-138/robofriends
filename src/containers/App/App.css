.App {
  text-align: center;
}

.title {
  font-family: 'Alumni Sans Inline One', cursive;
  margin-block: 3vh;
}