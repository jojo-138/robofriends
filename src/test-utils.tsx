import React, { PropsWithChildren } from 'react'
import { render } from '@testing-library/react'
import type { RenderOptions } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import type { AppStore } from './store'
import robotsReducer from './features/fetchRobots/robotsSlice';
import searchFieldReducer from './features/setSearchField/searchField';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  store?: AppStore;
};

export function renderWithProviders(
  ui: React.ReactElement,
  {
    store = configureStore({ reducer: { searchField: searchFieldReducer, robots: robotsReducer } }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>
  };
  
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions })};
};
