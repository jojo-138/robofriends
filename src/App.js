import { useEffect, useState } from 'react';
import './App.css';
import 'tachyons';
import { SearchBar } from './components/SearchBar';
import { CardList } from './components/CardList';

function App() {
  const [users, setUsers] = useState([]);
  useEffect (() => {
    fetch('https://jsonplaceholder.typicode.com/users')
      .then(res => res.json())
      .then(data => setUsers(data))
  }, [])

  const [input, setInput] = useState('');

  return (
    <div className="App">
      <div className='title f1 black-70'>Robofriends</div>
      <SearchBar handleChange={e => setInput(e.target.value)}/>
      <CardList users={users} input={input}/>
    </div>
  );
}

export default App;
