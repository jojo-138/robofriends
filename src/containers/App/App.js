import { connect } from 'react-redux';
import { setSearchField, fetchRobots } from '../../actions/actions';
import Main from '../Main/Main';
import './App.css';

const mapStateToProps = state => {
  const { searchField } = state.searchField;
  const { robots, isPending, fetchError } = state.robots;
  return {
    searchField: searchField,
    robots: robots,
    isPending: isPending,
    fetchError: fetchError
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onSearchChange: e => dispatch(setSearchField(e.target.value)),
    onFetchRobots: () => dispatch(fetchRobots())
  }
}

const App = ({ searchField, onSearchChange, robots, onFetchRobots }) => {
  return (
    <div className="App">
      <Main onFetchRobots={onFetchRobots} onSearchChange={onSearchChange} robots={robots} searchField={searchField} />
    </div>
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(App);