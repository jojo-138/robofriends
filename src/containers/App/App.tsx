import './App.css';
import 'tachyons';
import SearchBar from '../../components/SearchBar/SearchBar';
import CardList from '../../components/CardList/CardList';
import ErrorBoundary from '../../components/ErrorBoundary/ErrorBoundary';

const App = () => {
  return (
    <div className='App'>
      <div className='title f1 black-70'>Robofriends</div>
      <SearchBar />
      <ErrorBoundary>
        <CardList />
      </ErrorBoundary>
    </div>
  );
};

export default App;
