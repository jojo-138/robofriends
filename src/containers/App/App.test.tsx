import App from './App';
import { renderWithProviders } from "../../test-utils";
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const server = setupServer(
  rest.get('https://jsonplaceholder.typicode.com/users', (req, res, ctx) => {
    return res(
      ctx.json([
        {
          id: 1,
          name: 'john',
          email: 'john@gmail.com'
        },
        {
          id: 2,
          name: 'peter',
          email: 'peter@gmail.com'
        }
      ])
    );
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe('App component', () => {
  test('App renders correctly',() => {
    expect(renderWithProviders(<App />)).toMatchSnapshot();
  });
  
  test('robots filter correctly', async () => {
    renderWithProviders(<App />);

    await screen.findAllByRole('heading');

    userEvent.type(screen.getByRole('textbox'), 'john');

    expect(screen.getByText('john')).toBeInTheDocument();
    expect(screen.queryByText('peter')).not.toBeInTheDocument();
  });
});