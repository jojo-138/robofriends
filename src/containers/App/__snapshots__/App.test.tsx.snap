// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="title f1 black-70"
        >
          Robofriends
        </div>
        <div>
          <input
            aria-label="search robofriends"
            class="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
            placeholder="Search robofriends"
            type="text"
            value=""
          />
        </div>
        <div
          class="container bn pa3 f7 ma3 vh-75 overflow-y-auto"
          data-testid="card-list"
        >
          Fetching robofriends...
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="title f1 black-70"
      >
        Robofriends
      </div>
      <div>
        <input
          aria-label="search robofriends"
          class="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
          placeholder="Search robofriends"
          type="text"
          value=""
        />
      </div>
      <div
        class="container bn pa3 f7 ma3 vh-75 overflow-y-auto"
        data-testid="card-list"
      >
        Fetching robofriends...
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
